server/tests/unit/middleware/validation/date.test.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import httpMocks from 'node-mocks-http';
+import sinon from 'sinon';
+
+import * as mockedMerchantAuth from '../../../../src/middleware/__mocks__/merchant-auth';
+import * as merchantAuth from '../../../../src/middleware/merchant-auth';
+import validateAndFormatDate from '../../../../src/middleware/validation/date';
+import {BadRequestError} from '../../../../src/utils/http-errors';
+
+// Mock middlewares
+jest.mock('../../../../src/middleware/merchant-auth');
+
+const {mockMerchantAuth} = merchantAuth as typeof mockedMerchantAuth;
+
+describe('Date validation', () => {
+  beforeAll(() => {
+    mockMerchantAuth();
+  });
+
+  const validListing = {
```

Comment:
Why not take this from listingFixtures?

Reply:
@hzxcaryn Oh ya, I didn't realise I could use that.